server/src/main/java/org/spine3/server/command/EventIdSequence.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.command;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import org.spine3.base.CommandId;
+import org.spine3.base.EventId;
+
+import static org.spine3.base.Identifiers.idToString;
+
+/**
+ * Generates a sequence of identifiers of events produced in response to a command.
+ *
+ * @author Alexander Yevsyukov
+ * @see EventId
+ */
+class EventIdSequence {
+
+    /**
+     * The separator between command ID part and sequence number part.
+     */
+    @VisibleForTesting
+    static final char SEPARATOR = '-';
+
+    /**
+     * The radix for sequence number output.
+     */
+    private static final int RADIX = Character.MAX_RADIX;
+
+    /**
+     * The maximum number of events that fit into one digit sequence number.
+     */
+    static final int MAX_ONE_DIGIT_SIZE = RADIX;
+
+    /**
+     * The character appearing before the sequence number if the sequence size is bigger
+     * than {@link #RADIX}.
+     */
+    @VisibleForTesting
+    static final char LEADING_ZERO = '0';
+
+    /**
+     * The prefix with the command ID.
+     */
+    private final String commandIdPrefix;
+
+    /**
+     * The length of the sequence number suffix.
+     */
+    private final int suffixLength;
+
+    /**
+     * The number of the next ID in the sequence.
+     */
+    private int count = 0;
```

Comment:
Just in case, I'd use `AtomicInteger` instead. 

I realise our current use cases operate with `Event` ID generation in a synchronous manner. However, in future that may easily change.

Perfomance-wise it IS going to affect us; however, [`Strings.padStart` method](https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Strings.java#L88) already takes much more CPU and memory, than increments based upon `AtomicInteger` will.